pkg/apis/network/v1/egressrouter_types.go

```diff
@@ -0,0 +1,273 @@
+
+	// +kubebuilder:validation:Optional
+	// +optional
+	Status EgressRouterStatus `json:"status,omitempty"`
+}
+
+// EgressRouterSpec contains the configuration for an egress router.
+// Mode, networkInterface and addresses fields must be specified along with exactly one "Config" that matches the mode.
+// Each config consists of parameters specific to that mode.
+// +k8s:openapi-gen=true
+// +kubebuilder:validation:Required
+type EgressRouterSpec struct {
+	// Mode depicts the mode that is used for the egress router. The default mode is "Redirect" and is the only supported mode currently.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum="Redirect"
+	// +kubebuilder:default:="Redirect"
+	Mode EgressRouterMode `json:"mode"`
+
+	// Redirect represents the configuration parameters specific to redirect mode.
+	// +kubebuilder:validation:Optional
+	Redirect *RedirectConfig `json:"redirect,omitempty"`
+
+	// Specification of interface to create/use. The default is macvlan.
+	// Currently only macvlan is supported.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinProperties:=1
+	// +kubebuilder:validation:MaxProperties:=1
+	// +kubebuilder:default:={macvlan: {mode: Bridge}}
+	NetworkInterface EgressRouterInterface `json:"networkInterface"`
+
+	// List of IP addresses to configure on the pod's secondary interface.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems:=1
+	Addresses []EgressRouterAddress `json:"addresses"`
+}
+
+// EgressRouterMode defines the different types of modes that are supported for the egress router interface.
+// The default mode is "Redirect" and is the only supported mode currently.
+type EgressRouterMode string
+
+const (
+	// EgressRouterModeRedirect creates an egress router that sets up iptables rules to redirect traffic
+	// from its own IP address to one or more remote destination IP addresses.
+	EgressRouterModeRedirect EgressRouterMode = "Redirect"
+)
+
+// RedirectConfig represents the configuration parameters specific to redirect mode.
+type RedirectConfig struct {
+	// List of L4 parameters that define the destinations that the pod will connect to in redirect mode.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:MinItems:=1
+	RedirectRules []L4RedirectRule `json:"redirectRules,omitempty"`
+
+	// FallbackIP specifies the remote destination's IP address. Can be IPv4 or IPv6.
+	// If no redirect rules are specified, all traffic from the router are redirected to this IP.
+	// If redirect rules are specified, then any connections on any other port (undefined in the rules) on the router will be redirected to this IP.
+	// If redirect rules are specified and no fallback IP is provided, connections on other ports will simply be rejected.
+	// +kubebuilder:validation:Optional
+	FallbackIP CIDR `json:"fallbackIP,omitempty"`
+}
+
+// L4RedirectRule consists of L4parameters used to define a single remote destination in the redirect mode.
```

Comment:
Wording suggestion:

"L4RedirectRule defines a DNAT redirection from a given port to a destination IP and port."